[NNC] ported over sum and softmax to new scheme

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56718 [WIP] Added skeleton of python dispatch key
* #57426 [nnc] Added NNC lowerings for t/transpose/permute/expand + other cleaning
* #57427 [FX] Changes in order to move python key out of tree
* #56875 [nnc] ported conv2d lowering over
* **#56775 [NNC] ported over sum and softmax to new scheme**

Differential Revision: [D28173905](https://our.internmc.facebook.com/intern/diff/D28173905)

Chillee committed:
ported over sum

[ghstack-poisoned]

Chillee committed:
Update on "ported over sum"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"




[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good to me! I'm assuming tests pass and logic stays semantically the same.

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"


Differential Revision: [D28173905](https://our.internmc.facebook.com/intern/diff/D28173905)

[ghstack-poisoned]

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"


Differential Revision: [D28173905](https://our.internmc.facebook.com/intern/diff/D28173905)

[ghstack-poisoned]

Chillee commented:
@chillee has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28173905).

Chillee committed:
Update on "[NNC] ported over sum and softmax to new scheme"


Differential Revision: [D28173905](https://our.internmc.facebook.com/intern/diff/D28173905)

[ghstack-poisoned]

Chillee commented:
@chillee has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28173905).
